conan/tools/cmake/cmakedeps/cmakedeps.py

```diff
@@ -0,0 +1,54 @@
+import os
+
+from conan.tools.cmake.cmakedeps.templates.config import ConfigTemplate
+from conan.tools.cmake.cmakedeps.templates.config_version import ConfigVersionTemplate
+from conan.tools.cmake.cmakedeps.templates.macros import MacrosTemplate
+from conan.tools.cmake.cmakedeps.templates.target_configuration import TargetConfigurationTemplate
+from conan.tools.cmake.cmakedeps.templates.target_data import ConfigDataTemplate
+from conan.tools.cmake.cmakedeps.templates.targets import TargetsTemplate
+from conans.util.files import save
+
+
+class CMakeDeps(object):
+
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self.arch = self._conanfile.settings.get_safe("arch")
+        self.configuration = str(self._conanfile.settings.build_type)
+        self.configurations = [v for v in conanfile.settings.build_type.values_range if v != "None"]
+
+    def generate(self):
+        # Current directory is the generators_folder
+        generator_files = self.content
+        for generator_file, content in generator_files.items():
+            save(generator_file, content)
+
+    @property
+    def content(self):
+        macros = MacrosTemplate()
+        ret = {macros.filename: macros.render()}
+
+        host_requires = {r.ref.name: r for r in
+                         self._conanfile.dependencies.transitive_host_requires}
+        # Iterate all the transitive requires
+        for req in host_requires.values():
+
+            config_version = ConfigVersionTemplate(req, self.configuration)
```

Comment:
Consider passing to the components the full ``self`` pointing to ``CMakeDeps``, in case some more information or configurability is to be passed in the future.

Reply:
Did you finally consider it? You did thumbs-up, but no changes. I am fine to wait or not do it, just curious.